sdk/granter.go

```diff
@@ -27,16 +27,28 @@ const (
 	emptyTerm   = "DEFAULT"
 )
 
-// Represents a which KAS a split with the associated ID should shared with.
+// keySplitStep represents a which KAS a split with the associated ID should be shared with.
 type keySplitStep struct {
 	KAS, SplitID string
 }
 
-// Utility type to represent an FQN for an attribute.
+// AttributeNameFQN is a utility type to represent an FQN for an attribute.
 type AttributeNameFQN struct {
 	url, key string
 }
 
+func attributeURLPartsValid(parts []string) error {
+	for i, part := range parts {
+		if part == "" {
+			return fmt.Errorf("%w: empty url path parts are not allowed", ErrInvalid)
+		} else if i > 1 && // skip first two parts as they will have protocol slashes
+			strings.Contains(part, "/") {
+			return fmt.Errorf("%w: slash not allowed in name or values", ErrInvalid)
+		}
+	}
+	return nil
+}
```

Comment:
@jentfoo is the SDK panicking or the platform? 

When we add namespace grants we'll just want to make sure the namespace has a valid policy entry to ensure we can get the appropriate public key. Otherwise it's just a string comparison to derive the foreign key which will resolve to the KAS grant.

Also, we technically prescribe the `https://<ns>/attr/<attr-name>/value/<value-name>` as the format, but there's equally no reason why someone wouldn't want to use urn or similar.

Reply:
> is the SDK panicking or the platform?

I am still learning the code and how it's invoked.  I labeled the PR as `sdk` because all direct invocations are from within the SDK, but I don't know if this also effects the platform.

Reply:
I'll reach out tomorrow and show you our integration tests maybe you can help us identify how to do fuzz testing there 😄